examples/record-iteration/record_iteration.bal

```diff
@@ -0,0 +1,57 @@
+import ballerina/io;
+
+type Grades record {
+    int maths;
+    int physics;
+    int chemistry;
+    !...
+};
+
+public function main() {
+    // This creates a `Grades` record.
+    Grades grades = {maths: 80, physics: 75, chemistry: 65};
+
+    // The foreach variable is a 2-tuple which consists of the field name and the value of the field.
+    // The type of the field name is `string`. The variable type of the value depends on the types of fields of the record.
+    // If the fields and the rest descriptor are of same type (or if it is a closed record, the rest descriptor does not matter),
+    // the variable will also be of that same type. If not, the variable will be an `any` variable.
```

Comment:
How about something like
```ballerina
    // If the fields and the rest descriptor are of same type (the rest descriptor does not matter if it is a closed record),
    // the variable will also be of the same type. If not, the variable will be of type `any`.
```

Reply:
Wouldn't this be written as "REST descriptor" normally?

Reply:
@MaryamZi +1 will change.

@samgnaniah I think you may have got REST in the REST API sense mixed up with rest fields in open records. In an open record, a rest fields mean additional fields other than the ones defined in the descriptor.

Reply:
Understood. Thanks for the explanation.